### PR TITLE
Allow DBT 1.3.0

### LINKED ISSUE #85: upgrade to support dbt-core v1.3.0
## Background

The latest release cut for 1.3.0, `dbt-core==1.3.0rc2` was published on October 3, 2022 ([PyPI](https://pypi.org/project/dbt-core/1.3.0rc2/) | [Github](https://github.com/dbt-labs/dbt-core/releases/tag/v1.3.0rc2)). We are targeting releasing the official cut of 1.3.0 in time for the week of October 16 (in time for Coalesce conference).

We're trying to establish a following precedent w.r.t. minor versions:
Partner adapter maintainers release their adapter's minor version within four weeks of the initial RC being released. Given the delay on our side in notifying you, we'd like to set a target date of November 7 (four weeks from today) for maintainers to release their minor version

| Timeframe   | Date (intended) | Date (Actual) | Event                                                  |
| ----------- | --------------- | ------------- | ------------------------------------------------------ |
| D - 3 weeks | Sep 21          | Oct 10        | dbt Labs informs maintainers of upcoming minor release |
| D - 2 weeks | Sep 28          | Sep 28        | core 1.3 RC is released                                |
| Day D       | October 12      | Oct 12        | core 1.3 official is published                         |
| D + 2 weeks | October 26      | Nov 7         | dbt-adapter 1.3 is published                           |

## How to upgrade

https://github.com/dbt-labs/dbt-core/discussions/6011 is an open discussion with more detailed information, and https://github.com/dbt-labs/dbt-core/issues/6040 is for keeping track of the community's progress on releasing 1.2.0

Below is a checklist of work that would enable a successful 1.2.0 release of your adapter.

- [ ] Python Models (if applicable)
- [ ] Incremental Materialization: cleanup and standardization
- [ ] More functional adapter tests to inherit

### PR DESCRIPTION
**Pull request by pixie79**
resolves #85 

### Description

This updates setup.py to support a min and max dbt-core/dbt-spark version allowing us to support version. 1.2.0 and 1.3.0 for a period.

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-glue next" section.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -28,8 +28,9 @@
     long_description = f.read()
 
 package_name = "dbt-glue"
-package_version = "0.2.14"
-dbt_version = "1.2.0"
+package_version = "0.2.15"
+dbt_min_version = "1.2.0"
+dbt_max_version = "1.3.2"
```
Comment (nicor88) [inline setup.py:33]: why that? do you want to maintain cross compatibility? I would rather suggest to kept it easy and just add `dbt_version = "1.3.0"`

**Review by nicor88 — COMMENTED** [review]
See my comments, also update the CHANGELOG.md with your changes please.

File: setup.py (hunk 1 above)
Comment (pixie79) [reply setup.py:33]: I did that as currently it is compatible with both versions. So if someone does not yet want to update they can have the lower version of dbt-core. At the point it become incompatible then we should update. 
The other option which may be more contentious would be to bump the version to 1.3.0 like the main dbt packages do to show that this branch follows the 1.3 version of dbt-core?

File: setup.py [hunk 2]
```diff
@@ -28,9 +28,8 @@
     long_description = f.read()
 
 package_name = "dbt-glue"
-package_version = "0.2.15"
-dbt_min_version = "1.2.0"
-dbt_max_version = "1.3.2"
+package_version = "1.3.0"
```
Comment (nicor88) [inline setup.py:31]: personal opinion, this is quite a big bump that is not motivating the increase the major version.
How about using 0.3.0? I guess that quite some other features need to be released before we reach a total final stable version 1.0.0. Also following the semantic version, I will suggest increasing only the minor version, and keeping still the major to 0.

Comment (pixie79) [reply setup.py]: agreed

**Pull request merged**